### PR TITLE
[Yaml] Improve newline handling in folded scalar blocks

### PR DESCRIPTION
**Pull request by teohhanhui**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #15477 |
| License | MIT |
| Doc PR | N/A |

### CODE REVIEW

**Comment by javiereguiluz** [comment]
@teohhanhui I'm glad that after the discussion in #15477 issue you have created the pull request according to the format required by Symfony.

We have a bot called "fabbot" which reviews the submitted pull requests for basic errors. Although the bot is reporting an error for this pull request, we can safely ignore it because it's related to some feature needed for the unit tests.

So everything is ready to be reviewed by our team and ultimately merge it in Symfony. Let's wait for the revision. Thanks!

**Comment by teohhanhui** [comment]
@javiereguiluz No worries. Actually, I'm not new to Symfony and the contributing process.

**Comment by stof** [comment]
@javiereguiluz we should make fabbot ignore the Yaml test fixtures to avoid reporting this issue here (or we should refactor the way we pass Yaml test fixtures)

File: src/Symfony/Component/Yaml/Parser.php [hunk 1]
```diff
@@ -20,7 +20,7 @@
  */
 class Parser
 {
-    const FOLDED_SCALAR_PATTERN = '(?P<separator>\||>)(?P<modifiers>\+|\-|\d+|\+\d+|\-\d+|\d+\+|\d+\-)?(?P<comments> +#.*)?';
+    const BLOCK_SCALAR_PATTERN = '(?P<separator>\||>)(?P<modifiers>\+|\-|\d+|\+\d+|\-\d+|\d+\+|\d+\-)?(?P<comments> +#.*)?';
```
Comment (stof) [inline src/Symfony/Component/Yaml/Parser.php]: renaming a constant is a BC break

File: src/Symfony/Component/Yaml/Parser.php [hunk 2]
```diff
@@ -437,15 +437,15 @@ private function parseValue($value, $exceptionOnInvalidType, $objectSupport)
...
     /**
-     * Parses a folded scalar.
+     * Parses a block scalar.
      *
-     * @param string $separator   The separator that was used to begin this folded scalar (| or >)
-     * @param string $indicator   The indicator that was used to begin this folded scalar (+ or -)
-     * @param int    $indentation The indentation that was used to begin this folded scalar
+     * @param string $separator   The separator that was used to begin this block scalar (| or >)
+     * @param string $chomping    The chomping indicator that was used to begin this block scalar (+ or -)
+     * @param int    $indentation The indentation indicator that was used to begin this block scalar
```
Comment (stof) [inline src/Symfony/Component/Yaml/Parser.php]: Why indicator in this comment ?

Comment (teohhanhui) [inline src/Symfony/Component/Yaml/Parser.php]: [8.1.1.1. Block Indentation Indicator](http://www.yaml.org/spec/1.2/spec.html#id2793979)

File: src/Symfony/Component/Yaml/Parser.php (hunk 1 above)
Comment (teohhanhui) [inline src/Symfony/Component/Yaml/Parser.php]: What do you suggest then? Can I recreate it like this?

```php
const FOLDED_SCALAR_PATTERN = self::BLOCK_SCALAR_PATTERN;
```

File: src/Symfony/Component/Yaml/Parser.php (hunk 2 above)
Comment (teohhanhui) [inline src/Symfony/Component/Yaml/Parser.php]: I also think that `$separator` should be renamed `$style` (i.e. style indicator)

**Comment by teohhanhui** [comment]
PHP7 build is failing on Travis... Does it have anything to do with my code?

**Comment by jakzal** [comment]
Tests for [HttpFoundation fail](https://travis-ci.org/symfony/symfony/jobs/75031699#L1137), so it doesn't seem to be an issue related to your PR @teohhanhui.

**Comment by teohhanhui** [comment]
Are we good on this?

**Comment by teohhanhui** [comment]
ping @stof

**Comment by teohhanhui** [comment]
Any changes required before this will get merged?

**Comment by stof** [comment]
:+1:

**Comment by jakzal** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @teohhanhui.

**Pull request merged**